extensions/rvadapter/src/main/java/com/google/android/agera/rvadapter/RepositoryPresenterCompilerStates.java

```diff
@@ -69,6 +69,19 @@
   }
 
   /**
+   * Compiler state to specify how to generate stable IDs when
+   * {@link android.support.v7.widget.RecyclerView.Adapter#setHasStableIds(boolean)} is true.
+   */
+  interface RPStableId<TVal, TRet> {
+
+    /**
+     * Specifies a {@link Function} providing a stable id for the given model object instance.
```

Comment:
I'd prefer copying (a part of) the SLA from RepositoryPresenter.getItemId() here, just to drive home the requirement that the Function should be aware of other RepositoryPresenters that will be potentially sharing the same RepositoryAdapter when it comes to generating stable IDs.